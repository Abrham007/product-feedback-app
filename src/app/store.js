import { configureStore } from "@reduxjs/toolkit";

import productRequestsReducer from "../features/productRequests/productRequestsSlice";

export default configureStore({
  reducer: {
    productRequests: productRequestsReducer,
  },
});
