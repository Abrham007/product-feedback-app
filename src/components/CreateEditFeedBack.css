.CreateEditFeedBack {
  position: relative;
  width: 540px;
  display: flex;
  gap: 68px;
  flex-direction: column;
  background: #f7f8fd;
  margin: 92px 0px;
}

.CreateEditFeedBack__link {
  width: max-content;
  display: flex;
  gap: 15px;
  align-items: center;
  justify-content: flex-start;
  border: none;
  outline: none;
  flex-shrink: 0;
  border-radius: 10px;
  padding: 0px;
  color: #647196;
  font-size: 0.875rem;
  font-weight: 700;
  text-decoration: none;
  background-color: transparent;
}

.CreateEditFeedBack__link:hover {
  cursor: pointer;
  text-decoration: underline;
}

.CreateEditFeedBack__logo {
  position: absolute;
  top: 54px;
  left: 42px;
}

.CreateEditFeedBack__form {
  display: flex;
  flex-direction: column;
  gap: 35px;
  padding: 52px 42px 40px;
  border-radius: 10px;
  background: #fff;
}

.CreateEditFeedBack__header {
  color: #3a4374;
  font-size: 1.5rem;
  font-weight: 700;
  letter-spacing: -0.02081rem;
}

.CreateEditFeedBack__input-section {
  display: flex;
  flex-direction: column;
  gap: 24px;
}

.CreateEditFeedBack__input-box {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.CreateEditFeedBack__label {
  display: flex;
  flex-direction: column;
  gap: 2px;
}

.CreateEditFeedBack__sub-header {
  color: #3a4374;
  font-size: 0.875rem;
  font-weight: 700;
  letter-spacing: -0.01213rem;
}

.CreateEditFeedBack__detail {
  color: #647196;
  font-size: 0.875rem;
  font-weight: 400;
}

.CreateEditFeedBack__input {
  border: none;
  outline: none;
  padding: 13px 24px;
  border-radius: 5px;
  background: #f7f8fd;
  font-family: Jost;
  color: #3a4374;
  font-size: 0.9375rem;
  font-weight: 400;
}

.CreateEditFeedBack__input:hover,
.CreateEditFeedBack__input:focus {
  cursor: pointer;
  border: 1px solid #4661e6;
}

.CreateEditFeedBack__input--select {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 456px;
  height: 48px;
}

.CreateEditFeedBack__input--text-area {
  resize: none;
  height: 96px;
}

.CreateEditFeedBack__btn-box {
  display: flex;
  gap: 16px;
  justify-content: flex-end;
}

.CreateEditFeedBack__btn {
  border: none;
  outline: none;
  padding: 12px 24px;
  color: #f2f4fe;
  font-size: 0.875rem;
  font-weight: 700;
  border-radius: 10px;
}

.CreateEditFeedBack__btn--delete {
  margin-right: auto;
  background: #d73737;
}

.CreateEditFeedBack__btn--delete:hover {
  cursor: pointer;
  background: #e98888;
}

.CreateEditFeedBack__btn--cancel {
  background: #3a4374;
}

.CreateEditFeedBack__btn--cancel:hover {
  cursor: pointer;
  background: #656ea3;
}

.CreateEditFeedBack__btn--add {
  background: #ad1fea;
}

.CreateEditFeedBack__btn--add:hover {
  cursor: pointer;
  background: #c75af6;
}

.error {
  color: #d73737;
  font-size: 0.875rem;
  font-weight: 400;
}

@media screen and (max-width: 768px) {
  .CreateEditFeedBack {
    width: 327px;
    gap: 55px;
  }

  .CreateEditFeedBack__form {
    gap: 24px;
  }

  .CreateEditFeedBack__logo {
    width: 40px;
    height: 40px;
  }

  .CreateEditFeedBack__header {
    font-size: 1.125rem;
  }

  .CreateEditFeedBack__sub-header,
  .CreateEditFeedBack__detail {
    font-size: 0.8125rem;
  }

  .CreateEditFeedBack__btn-box {
    margin-top: 16px;
    flex-direction: column-reverse;
    align-items: stretch;
  }

  .CreateEditFeedBack__btn--delete {
    width: 100%;
  }
}
