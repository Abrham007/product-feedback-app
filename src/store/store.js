import { configureStore } from "@reduxjs/toolkit";
import currentUserReducer from "../features/currentUser/currentUserSlice";
import productRequestsReducer from "../features/productRequests/productRequestsSlice";

export default configureStore({
  reducer: {
    currentUser: currentUserReducer,
    productRequests: productRequestsReducer,
  },
});
