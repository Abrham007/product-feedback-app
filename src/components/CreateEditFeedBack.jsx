import "./CreateEditFeedBack.css";
import { useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import arrowLeftIcon from "../assets/shared/icon-arrow-left.svg";
import newFeedbackIcon from "../assets/shared/icon-new-feedback.svg";
import editFeedbackIcon from "../assets/shared/icon-edit-feedback.svg";
import { Link, useParams, useNavigate } from "react-router-dom";
import FeedBackInput from "./FeedBackInput";

function CreateEditFeedBack(props) {
  let { id } = useParams();
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    reset,
    formState,
    formState: { errors, isSubmitSuccessful },
  } = useForm();

  const [currentCategory, setCurrentCategory] = useState(0);
  const [currentStatus, setCurrentStatus] = useState(0);
  let categoryList = ["Feature", "UI", "UX", "Enhancement", "Bug"];
  let statusList = ["Suggestion", "Planned", "In-Progress", "Live"];
  let CURRENTFEEDBACK = props.appData?.productRequests.find(
    (req) => req.id == id
  );

  function handleCurrentCategory(categoryIndex) {
    console.log("somflds");
    setCurrentCategory(categoryIndex);
  }

  function handleCurrentStatus(statusIndex) {
    setCurrentStatus(statusIndex);
  }

  function addData(data) {
    props.handleAppData((prevValue) => {
      let tempValue = { ...prevValue };
      tempValue.productRequests.push({
        id: prevValue.productRequests.length + 1,
        title: data.title,
        category: categoryList[currentCategory].toLowerCase(),
        upvotes: 0,
        status: statusList[currentStatus].toLowerCase(),
        description: data.description,
      });
      return tempValue;
    });
  }

  function editData(data) {
    props.handleAppData((prevValue) => {
      let oldFeedback = prevValue.productRequests.find((req) => req.id == id);
      let feedbackIndex = prevValue.productRequests.findIndex(
        (req) => req.id == id
      );
      let newFeedback = {
        id: oldFeedback.id,
        title: data.title,
        category: categoryList[currentCategory].toLowerCase(),
        upvotes: oldFeedback.upvotes,
        status: statusList[currentStatus].toLowerCase(),
        description: data.description,
      };
      if (oldFeedback.comments) {
        newFeedback.comments = oldFeedback.comments;
      }
      let tempValue = { ...prevValue };
      tempValue.productRequests[feedbackIndex] = newFeedback;
      return tempValue;
    });
  }

  function handleDelete(event) {
    event.preventDefault();
    props.handleAppData((prevValue) => {
      let feedbackIndex = prevValue.productRequests.findIndex(
        (req) => req.id == id
      );
      let tempValue = { ...prevValue };
      tempValue.productRequests.splice(feedbackIndex, 1);
      return tempValue;
    });
    navigate("/");
  }

  function onSubmit(data) {
    if (props.appData) {
      editData(data);
      navigate(`/feedbackdetail/${id}`);
    } else {
      addData(data);
      navigate("/");
    }
  }

  useEffect(() => {
    if (props.appData) {
      setCurrentCategory((prevVaule) => {
        let index = categoryList.findIndex(
          (category) => category.toLowerCase() == CURRENTFEEDBACK.category
        );
        return index;
      });

      setCurrentStatus((prevVaule) => {
        let index = statusList.findIndex(
          (status) => status.toLowerCase() == CURRENTFEEDBACK.status
        );
        return index;
      });
    }
  }, []);

  useEffect(() => {
    if (isSubmitSuccessful) {
      reset({ title: "", description: "" });
    }
  }, [formState]);
  return (
    <div className="CreateEditFeedBack">
      <button onClick={() => navigate(-1)} className="CreateEditFeedBack__link">
        <img src={arrowLeftIcon} alt=""></img>
        Go Back
      </button>
      <img
        src={props.appData ? editFeedbackIcon : newFeedbackIcon}
        alt=""
        height={56}
        width={56}
        className="CreateEditFeedBack__logo"
      ></img>
      <form
        onSubmit={handleSubmit(onSubmit)}
        className="CreateEditFeedBack__form"
      >
        <h2 className="CreateEditFeedBack__header">
          {props.appData
            ? `Editing ‘${CURRENTFEEDBACK.title}’`
            : "Create New Feedback"}
        </h2>
        <div className="CreateEditFeedBack__input-section">
          <FeedBackInput
            register={register}
            errors={errors}
            name={"title"}
            title={"Feedback Title"}
            description={"Add a short, descriptive headline"}
            type={"text"}
            defaultValue={props.appData ? CURRENTFEEDBACK.title : ""}
          />

          <FeedBackInput
            name={"category"}
            title={"Category"}
            description={"Choose a category for your feedback"}
            type={"select"}
            selectList={categoryList}
            handleCurrent={handleCurrentCategory}
            selectedIndex={currentCategory}
          />

          {props.appData && (
            <FeedBackInput
              name={"status"}
              title={"Update Status"}
              description={"Change feedback state"}
              type={"select"}
              selectList={statusList}
              handleCurrent={handleCurrentStatus}
              selectedIndex={currentStatus}
            />
          )}

          <FeedBackInput
            register={register}
            errors={errors}
            name={"description"}
            title={"Feedback Detail"}
            description={
              "Include any specific comments on what should be improved, added, etc."
            }
            type={"text-area"}
            defaultValue={props.appData ? CURRENTFEEDBACK.description : ""}
          />
        </div>
        <div className="CreateEditFeedBack__btn-box">
          {props.appData && (
            <button
              onClick={handleDelete}
              className="CreateEditFeedBack__btn CreateEditFeedBack__btn--delete"
            >
              Delete
            </button>
          )}
          <button
            onClick={() => navigate(-1)}
            className="CreateEditFeedBack__btn CreateEditFeedBack__btn--cancel"
          >
            Cancel
          </button>
          <button
            type="submit"
            className="CreateEditFeedBack__btn CreateEditFeedBack__btn--add"
          >
            {props.appData ? "Save Changes" : "Add Feedback"}
          </button>
        </div>
      </form>
    </div>
  );
}

export default CreateEditFeedBack;
