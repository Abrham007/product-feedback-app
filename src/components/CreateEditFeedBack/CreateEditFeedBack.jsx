import "./CreateEditFeedBack.css";
import { useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import arrowLeftIcon from "../../assets/shared/icon-arrow-left.svg";
import newFeedbackIcon from "../../assets/shared/icon-new-feedback.svg";
import editFeedbackIcon from "../../assets/shared/icon-edit-feedback.svg";
import { Link, useParams, useNavigate } from "react-router-dom";
import FeedBackInput from "../UI/Inputs/FeedBackInput";
import { useDispatch, useSelector } from "react-redux";
import BackBtn from "../UI/Buttons/BackBtn";
import {
  addNewPost,
  editPost,
  deletePost,
  selectProductRequests,
} from "../../features/productRequests/productRequestsSlice";

function CreateEditFeedBack(props) {
  // The id passed for editing from the FeedBackDeatil component
  let { id } = useParams();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const {
    register,
    handleSubmit,
    reset,
    formState,
    formState: { errors, isSubmitSuccessful },
  } = useForm();

  // This is for the custom select input
  const [currentCategory, setCurrentCategory] = useState(0);
  const [currentStatus, setCurrentStatus] = useState(0);

  let categoryList = ["Feature", "UI", "UX", "Enhancement", "Bug"];
  let statusList = ["Suggestion", "Planned", "In-Progress", "Live"];

  // This will get the post for editing for better UX from the redux store
  let CURRENTFEEDBACK = useSelector((state) =>
    state.productRequests.posts.find((req) => req._id == id)
  );

  // This is passed to the custom select input's
  function handleCurrentCategory(categoryIndex) {
    setCurrentCategory(categoryIndex);
  }

  function handleCurrentStatus(statusIndex) {
    setCurrentStatus(statusIndex);
  }

  // This will add new post directly to the database through redux
  async function addData(data) {
    try {
      dispatch(
        addNewPost({
          title: data.title,
          category: categoryList[currentCategory].toLowerCase(),
          upvotes: 0,
          status: statusList[currentStatus].toLowerCase(),
          description: data.description,
        })
      );
    } catch (err) {
      console.log("Failed to save the post: ", err);
    }
  }

  // This will edit old post directly to the database through redux
  async function editData(data) {
    try {
      dispatch(
        editPost({
          postId: id,
          title: data.title,
          category: categoryList[currentCategory].toLowerCase(),
          status: statusList[currentStatus].toLowerCase(),
          description: data.description,
        })
      );
    } catch (err) {
      console.error("Failed to edit the post: ", err);
    }
  }

  // This will delelte a post directly on the database through redux
  function handleDelete(event) {
    event.preventDefault();
    try {
      dispatch(
        deletePost({
          postId: id,
        })
      );
    } catch {}

    navigate("/");
  }

  // This will determine to edit post or create the post and navigate
  // to the respective pages
  function onSubmit(data) {
    if (props.isEdit) {
      editData(data);
      navigate(`/feedbackdetail/${id}`);
    } else {
      addData(data);
      navigate("/");
    }
  }

  // This will load the correct catagory and status if it is on edit
  // from the redux store
  useEffect(() => {
    if (props.isEdit) {
      let currentCategoryIndex = categoryList.findIndex(
        (category) => category.toLowerCase() == CURRENTFEEDBACK.category
      );
      let currentStatusIndex = statusList.findIndex(
        (status) => status.toLowerCase() == CURRENTFEEDBACK.status
      );
      setCurrentCategory(currentCategoryIndex);
      setCurrentStatus(currentStatusIndex);
    }
  }, []);

  // This will reset the input fields upon successful submtion
  useEffect(() => {
    if (isSubmitSuccessful) {
      reset({ title: "", description: "" });
    }
  }, [formState]);
  return (
    <div className="CreateEditFeedBack">
      <BackBtn />

      <img
        src={props.isEdit ? editFeedbackIcon : newFeedbackIcon}
        alt=""
        height={56}
        width={56}
        className="CreateEditFeedBack__logo"
      ></img>

      <form
        role="form"
        onSubmit={handleSubmit(onSubmit)}
        className="CreateEditFeedBack__form"
      >
        <h2 className="CreateEditFeedBack__header">
          {props.isEdit
            ? `Editing ‘${CURRENTFEEDBACK.title}’`
            : "Create New Feedback"}
        </h2>

        <div className="CreateEditFeedBack__input-section">
          <FeedBackInput
            register={register}
            errors={errors}
            name={"title"}
            title={"Feedback Title"}
            description={"Add a short, descriptive headline"}
            type={"text"}
            defaultValue={props.isEdit ? CURRENTFEEDBACK.title : ""}
          />

          <FeedBackInput
            name={"category"}
            title={"Category"}
            description={"Choose a category for your feedback"}
            type={"select"}
            selectList={categoryList}
            handleCurrent={handleCurrentCategory}
            selectedIndex={currentCategory}
          />

          {props.isEdit && (
            <FeedBackInput
              name={"status"}
              title={"Update Status"}
              description={"Change feedback state"}
              type={"select"}
              selectList={statusList}
              handleCurrent={handleCurrentStatus}
              selectedIndex={currentStatus}
            />
          )}

          <FeedBackInput
            register={register}
            errors={errors}
            name={"description"}
            title={"Feedback Detail"}
            description={
              "Include any specific comments on what should be improved, added, etc."
            }
            type={"text-area"}
            defaultValue={props.isEdit ? CURRENTFEEDBACK.description : ""}
          />
        </div>

        <div className="CreateEditFeedBack__btn-box">
          {props.isEdit && (
            <button
              onClick={handleDelete}
              className="CreateEditFeedBack__btn CreateEditFeedBack__btn--delete"
            >
              Delete
            </button>
          )}

          <button
            onClick={() => navigate(-1)}
            className="CreateEditFeedBack__btn CreateEditFeedBack__btn--cancel"
          >
            Cancel
          </button>

          <button
            type="submit"
            className="CreateEditFeedBack__btn CreateEditFeedBack__btn--add"
          >
            {props.isEdit ? "Save Changes" : "Add Feedback"}
          </button>
        </div>
      </form>
    </div>
  );
}

export default CreateEditFeedBack;
