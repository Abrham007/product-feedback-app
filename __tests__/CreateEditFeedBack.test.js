import CreateEditFeedBack from "../src/components/CreateEditFeedBack";
import { Route, MemoryRouter, Routes, BrowserRouter } from "react-router-dom";
import { preloadedData } from "../utils/test-utils";
import { setupStore } from "../src/store/store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

describe("CreateEditFeedBack", () => {
  describe("on create", () => {
    const component = (
      <Provider store={setupStore(preloadedData)}>
        <BrowserRouter>
          <CreateEditFeedBack></CreateEditFeedBack>
        </BrowserRouter>
      </Provider>
    );
    test("renders the correct JSX", () => {
      let tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("on edit", () => {
    test("renders the correct JSX", () => {
      let component = renderer.create(
        <Provider store={setupStore(preloadedData)}>
          <MemoryRouter initialEntries={[`/edit/${1}`]}>
            <Routes>
              <Route
                path="/edit/:id"
                element={
                  <CreateEditFeedBack isEdit={true}></CreateEditFeedBack>
                }
              ></Route>
            </Routes>
          </MemoryRouter>
        </Provider>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
