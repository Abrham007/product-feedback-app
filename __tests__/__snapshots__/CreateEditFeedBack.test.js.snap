// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEditFeedBack on create renders the correct JSX 1`] = `
<div
  className="CreateEditFeedBack"
>
  <button
    className="BackBtn"
    onClick={[Function]}
    style={
      {
        "color": "#647196",
      }
    }
  >
    <svg
      height="10"
      width="7"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 9L2 5l4-4"
        fill="none"
        fillRule="evenodd"
        stroke="#4661E6"
        strokeWidth="2"
      />
    </svg>
    <span>
      Go Back
    </span>
  </button>
  <img
    alt=""
    className="CreateEditFeedBack__logo"
    height={56}
    width={56}
  />
  <form
    className="CreateEditFeedBack__form"
    onSubmit={[Function]}
    role="form"
  >
    <h2
      className="CreateEditFeedBack__header"
    >
      Create New Feedback
    </h2>
    <div
      className="CreateEditFeedBack__input-section"
    >
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor="title"
        >
          <h3
            className="FeedBackInput__header"
          >
            Feedback Title
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Add a short, descriptive headline
          </p>
        </label>
        <input
          className="FeedBackInput__input"
          defaultValue=""
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          style={{}}
        />
      </div>
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor={null}
        >
          <h3
            className="FeedBackInput__header"
          >
            Category
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Choose a category for your feedback
          </p>
        </label>
        <div
          className="FromSelect"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <span>
            Feature
          </span>
          <img
            alt=""
          />
        </div>
      </div>
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor="description"
        >
          <h3
            className="FeedBackInput__header"
          >
            Feedback Detail
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Include any specific comments on what should be improved, added, etc.
          </p>
        </label>
        <textarea
          className="FeedBackInput__input FeedBackInput__input--text-area"
          defaultValue=""
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          style={{}}
        />
      </div>
    </div>
    <div
      className="CreateEditFeedBack__btn-box"
    >
      <button
        className="CreateEditFeedBack__btn CreateEditFeedBack__btn--cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="CreateEditFeedBack__btn CreateEditFeedBack__btn--add"
        type="submit"
      >
        Add Feedback
      </button>
    </div>
  </form>
</div>
`;

exports[`CreateEditFeedBack on edit renders the correct JSX 1`] = `
<div
  className="CreateEditFeedBack"
>
  <button
    className="BackBtn"
    onClick={[Function]}
    style={
      {
        "color": "#647196",
      }
    }
  >
    <svg
      height="10"
      width="7"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 9L2 5l4-4"
        fill="none"
        fillRule="evenodd"
        stroke="#4661E6"
        strokeWidth="2"
      />
    </svg>
    <span>
      Go Back
    </span>
  </button>
  <img
    alt=""
    className="CreateEditFeedBack__logo"
    height={56}
    width={56}
  />
  <form
    className="CreateEditFeedBack__form"
    onSubmit={[Function]}
    role="form"
  >
    <h2
      className="CreateEditFeedBack__header"
    >
      Editing ‘Add tags for solutions’
    </h2>
    <div
      className="CreateEditFeedBack__input-section"
    >
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor="title"
        >
          <h3
            className="FeedBackInput__header"
          >
            Feedback Title
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Add a short, descriptive headline
          </p>
        </label>
        <input
          className="FeedBackInput__input"
          defaultValue="Add tags for solutions"
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          style={{}}
        />
      </div>
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor={null}
        >
          <h3
            className="FeedBackInput__header"
          >
            Category
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Choose a category for your feedback
          </p>
        </label>
        <div
          className="FromSelect"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <span>
            Feature
          </span>
          <img
            alt=""
          />
        </div>
      </div>
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor={null}
        >
          <h3
            className="FeedBackInput__header"
          >
            Update Status
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Change feedback state
          </p>
        </label>
        <div
          className="FromSelect"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <span>
            Suggestion
          </span>
          <img
            alt=""
          />
        </div>
      </div>
      <div
        className="FeedBackInput"
      >
        <label
          className="FeedBackInput__label"
          htmlFor="description"
        >
          <h3
            className="FeedBackInput__header"
          >
            Feedback Detail
          </h3>
          <p
            className="FeedBackInput__description"
          >
            Include any specific comments on what should be improved, added, etc.
          </p>
        </label>
        <textarea
          className="FeedBackInput__input FeedBackInput__input--text-area"
          defaultValue="Easier to search for solutions based on a specific stack."
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          style={{}}
        />
      </div>
    </div>
    <div
      className="CreateEditFeedBack__btn-box"
    >
      <button
        className="CreateEditFeedBack__btn CreateEditFeedBack__btn--delete"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="CreateEditFeedBack__btn CreateEditFeedBack__btn--cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="CreateEditFeedBack__btn CreateEditFeedBack__btn--add"
        type="submit"
      >
        Save Changes
      </button>
    </div>
  </form>
</div>
`;
